### PR TITLE
document the new ElectPrimaryPolicy and related items

### PR DESCRIPTION
**Pull request by ahgittin**
Documentation for https://github.com/apache/brooklyn-server/pull/879

This has been developed in the field but is now generalized to the point I think it will be useful for many cases.

@aledsage @grkvlt -- particularly welcome feedback from y'all as you've done similar things I know

### CODE REVIEW

File: guide/blueprints/_elect-primary-policies.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+
+There are a collection of policies, enrichers, and effectors to assist with common
+failover scenarios and more general anything which wants a primary elected and
```
Comment (drigodwin) [inline guide/blueprints/_elect-primary-policies.md]: general -> generally,

Comment (drigodwin) [reply guide/blueprints/_elect-primary-policies.md]: which wants a primary elected and re-elected. -> which requires the election and re-election of a primary member.

File: guide/blueprints/_elect-primary-policies.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+All the `primary.*` parameters accepted by that effector can be defined on the policy and will be passed to the effector, along with an `event` parameter indicating the sensor which triggered the election.
+
+The policy also accepts a `propagate.primary.sensors` list of strings or sensors.
+If present, this will add the `PropagatePrimaryEnricher` enricher with those sensors set to
+be propagated (but not effectors).
+For more sophisticated configuration, that enricher can be added and configured directly instead.  
+
+If no quorum.up or quorum.running is set on the entity, both will be set to a constant 1.
```
Comment (drigodwin) [inline guide/blueprints/_elect-primary-policies.md]: put backticks round `quorum.up` and `quorum.running`?

File: guide/blueprints/_elect-primary-policies.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+#### ElectPrimary Policy
+
+- org.apache.brooklyn.policy.ha.ElectPrimaryPolicy
+
+The ElectPrimaryPolicy acts to keep exactly one of its children or members as primary, promoting and demoting them when required.
+
+A simple use case is where we have two children, call them North and South, and we wish for North to be primary.  If North fails, however, we want to promote and fail over to South.  This can be done by:
+
+* adding this policy at the parent
+* setting ` ha.primary.weight` on North
```
Comment (drigodwin) [inline guide/blueprints/_elect-primary-policies.md]: erroneous space before ha

File: guide/blueprints/policies.md [hunk 4]
```diff
@@ -78,11 +78,18 @@ example of these working together.
...
+`ha.entityFailed` (or other configurable sensor) as typically emitted by the ServiceFailureDetector enricher.  
 The [introduction to policies]({{book.path.docs}}/start/policies.md) shows a worked 
 example of this policy in use.
 
 
+#### ServiceFailureDetector Enricher
+
+- org.apache.brooklyn.policy.ha.ServiceFailureDetector
+
+TODO
```
Comment (drigodwin) [inline guide/blueprints/policies.md]: I don't think we should include TODOs in the docs

**Comment by ahgittin** [comment]
thanks @drigodwin - all addressed

File: guide/blueprints/_elect-primary-policies.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+* Causing `promote` and `demote` effectors to be invoked on the appropriate nodes when the primary is elected/changed
+  (with the parent reset to `STARTING` while this occurs)
+* Mirroring sensors and optionally effectors from the primary to the parent
+
+A simple example is as follows, deploying two `item` entities with one designated as primary
+and its `main.uri` sensor published at the root. If "Preferred Item" fails, "Failover Item"
+will be become the primary. Any `demote` effector on "Preferred Item" and any `promote` effector
+on "Failover Item" will be invoked on failover.
+
+```
```
Comment (tbouron) [inline guide/blueprints/_elect-primary-policies.md]: Miss `yaml` after the 3 backticks

**Comment by ahgittin** [comment]
all comments addressed i think, thanks.  ready for mege.

**Pull request merged**